### PR TITLE
Add resource and service tests for engagement PATCH request

### PR DESCRIPTION
**Pull request by jadmsaadaot**
*Issue #:*
https://github.com/bcgov/met-public/issues/510

*Description of changes:*

- Add integration test to test PATCH endpoint
- Add unit test to test edit_engagement service method

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-api/tests/unit/services/test_engagement.py [hunk 1]
```diff
@@ -32,3 +33,16 @@ def test_create_engagement(session):  # pylint:disable=unused-argument
...
+
+
+def test_patch_engagement(session):  # pylint:disable=unused-argument
+    """Assert that an Org can be created."""
+    saved_engagament = factory_engagement_model()
+    engagement_edits = {
+        'id': saved_engagament.id,
+        'name': 'Updated engagement name'
+    }
+    assert engagement_edits['name'] != saved_engagament.name
```
Comment (saravanpa-aot) [inline met-api/tests/unit/services/test_engagement.py:45]: using faker to generate a random name can take away this assertion

File: met-api/tests/unit/services/test_engagement.py [hunk 2]
```diff
@@ -32,3 +33,16 @@ def test_create_engagement(session):  # pylint:disable=unused-argument
...
+    upadted_engagement = EngagementService().edit_engagement(engagement_edits)
+    # fetch the engagement with id and assert
+    assert upadted_engagement.name == engagement_edits['name']
```
Comment (saravanpa-aot) [inline met-api/tests/unit/services/test_engagement.py:48]: may be assert a few more thing , just to ensure other data didnt get affected.. we had issue with status_code being overwritten

File: met-api/tests/unit/api/test_engagement.py [hunk 3]
```diff
@@ -49,3 +49,23 @@ def test_get_engagements(client, jwt, session, engagement_info):  # pylint:disab
...
     assert created_eng.get('result').get('content') == rv.json.get('result').get('content')
+
+
+@pytest.mark.parametrize('engagement_info', [TestEngagementInfo.engagement1])
+def test_patch_engagement(client, jwt, session, engagement_info):  # pylint:disable=unused-argument
+    """Assert that an survey can be POSTed."""
+    headers = factory_auth_header(jwt=jwt, claims=TestJwtClaims.no_role)
+    engagement = factory_engagement_model()
+    engagement_id = str(engagement.id)
+    new_engagement_name = 'new_engagement_name'
```
Comment (saravanpa-aot) [inline met-api/tests/unit/api/test_engagement.py:60]: I would use faker.name() here as well..

File: met-api/tests/unit/api/test_engagement.py [hunk 4]
```diff
@@ -49,3 +49,23 @@ def test_get_engagements(client, jwt, session, engagement_info):  # pylint:disab
...
+    rv = client.patch('/api/engagements/', data=json.dumps({'id': engagement_id, 'name': new_engagement_name}),
+                      headers=headers, content_type='application/json')
+
+    assert rv.status_code == 200
```
Comment (saravanpa-aot) [inline met-api/tests/unit/api/test_engagement.py:64]: Does that patch returns something?May be assert that as well.Will be easier in the future incase if we are changing the patch response to something else

File: met-api/tests/unit/services/test_engagement.py (hunk 1 above)
Comment (jadmsaadaot) [reply met-api/tests/unit/services/test_engagement.py]: Done

File: met-api/tests/unit/services/test_engagement.py (hunk 2 above)
Comment (jadmsaadaot) [reply met-api/tests/unit/services/test_engagement.py]: done

File: met-api/tests/unit/api/test_engagement.py (hunk 3 above)
Comment (jadmsaadaot) [reply met-api/tests/unit/api/test_engagement.py]: done

File: met-api/tests/unit/api/test_engagement.py (hunk 4 above)
Comment (jadmsaadaot) [reply met-api/tests/unit/api/test_engagement.py:78]: done

**Review by saravanpa-aot — APPROVED** [review]

**Review by tiagograf-AOT — APPROVED** [review]

**Pull request merged**